crates/topos-tce-synchronizer/src/lib.rs

```diff
@@ -78,6 +92,104 @@ pub enum SynchronizerError {
 
     #[error("Unable to execute shutdown on the Synchronizer: {0}")]
     ShutdownCommunication(mpsc::error::SendError<oneshot::Sender<()>>),
+
+    #[error("No network protocol receiver set")]
+    NoProtocolReceiver,
 }
 
 pub enum SynchronizerEvent {}
+
+#[derive(Clone)]
+pub struct SynchronizerService {
+    pub validator_store: Arc<ValidatorStore>,
+}
+
+#[async_trait::async_trait]
+impl GrpcSynchronizerService for SynchronizerService {
+    async fn fetch_certificates(
+        &self,
+        request: Request<FetchCertificatesRequest>,
+    ) -> Result<Response<FetchCertificatesResponse>, Status> {
+        let request = request.into_inner();
+        let certificate_ids: Vec<CertificateId> = request
+            .certificates
+            .into_iter()
+            .map(|c| c.try_into().unwrap())
```

Comment:
Feels like we shouldn't panic here (didn't we fix a bug in cert id parsing last week?)